pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/PublishDraftVersion.java

```diff
@@ -114,46 +120,40 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
             if(canPublish) {
                 serviceResourceResolver = serviceResourceResolverProvider.getServiceResourceResolver();
             }
-            Module module = serviceResourceResolver.getResource(request.getResource().getPath()).adaptTo(Module.class);
+            Optional<? extends Document> document = getDocument(request, serviceResourceResolver);
             Locale locale = getLocale(request);
             String variant = getVariant(request);
             // Get the draft version, there should be one
-            Optional<ModuleVersion> versionToRelease = module.getDraftVersion(locale, variant);
+            Optional<? extends DocumentVersion> versionToRelease = document.get().getDraftVersion(locale, variant);
             if (!versionToRelease.isPresent()) {
                 response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
-                        "The module doesn't have a draft version to be released");
+                        "The document doesn't have a draft version to be released");
                 return;
             } else if (versionToRelease.get().metadata().getOrCreate().productVersion().get() == null
                     || versionToRelease.get().metadata().getOrCreate().productVersion().get().isEmpty()) {
                 // Check if productVersion is set
                 response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
                         "The version to be released doesn't have productVersion metadata");
                 return;
-            } else if (versionToRelease.get().metadata().getOrCreate().urlFragment().get() == null
-                    || versionToRelease.get().metadata().getOrCreate().urlFragment().get().isEmpty()) {
-                // Check if urlFragment is set
-                response.setStatus(HttpServletResponse.SC_PRECONDITION_FAILED,
-                        "The version to be released doesn't have urlFragment metadata");
-                return;
             } else {
                 // Draft becomes the new released version
-                ModuleVariant moduleVariant = traverseFrom(module)
-                        .toChild(m -> module.locale(locale))
-                        .toChild(ModuleLocale::variants)
+                DocumentVariant docVariant = traverseFrom(document.get())
+                        .toChild(m -> document.get().locale(locale))
```

Comment:
@rednitish I realize you did not write this code originally, but now that I'm looking at it, I see that the variable `m` here is just unused. That has me wondering if the initial call to `traverseFrom` is unnecessary, or if this technique isn't being used in the wrong way.

@carlosmunoz Since you're the mastermind behind most of the model api, can you comment on whether this is the best way to get from Document down to ultimately DocumentVariant? Am I right in thinking that the discarding of `m` is a code smell? And what's the difference between using ResourceTraversal vs using the model api directly, such as `document.get().locale(whatever).get().variants().get().variant("variant").get()`?